compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpServiceValidator.java

```diff
@@ -79,6 +93,47 @@ public void perform(SyntaxNodeAnalysisContext syntaxNodeAnalysisContext) {
         }
     }
 
+    private static void extractServiceAnnotationAndValidate(SyntaxNodeAnalysisContext ctx,
+                                                            ServiceDeclarationNode serviceDeclarationNode) {
+        Optional<MetadataNode> metadataNodeOptional = serviceDeclarationNode.metadata();
+
+        if (metadataNodeOptional.isEmpty()) {
+            return;
+        } else {
+            NodeList<AnnotationNode> annotations = metadataNodeOptional.get().annotations();
+            for (AnnotationNode annotation : annotations) {
+                Node annotReference = annotation.annotReference();
+                String annotName = annotReference.toString();
+                Optional<MappingConstructorExpressionNode> annotValue = annotation.annotValue();
+                if (annotReference.kind() == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+                    String[] annotStrings = annotName.split(SEMICOLON);
+                    if (SERVICE_CONFIG_ANNOTATION.equals(annotStrings[annotStrings.length - 1].trim())
+                            && (annotValue.isPresent())) {
+                        MappingConstructorExpressionNode mapping = annotValue.get();
+                        for (MappingFieldNode field : mapping.fields()) {
+                            String fieldName = field.toString();
+                            fieldName = fieldName.trim().replaceAll(UNNECESSARY_CHARS_REGEX, "");
+                            if (field.kind() == SyntaxKind.SPECIFIC_FIELD) {
+                                String[] strings = fieldName.split(SEMICOLON, 2);
```

Comment:
Can we use semantic model here? I think this string manipulations can be avoided

Reply:
Eg: Optional<Symbol> resourceMethodSymbolOptional = ctx.semanticModel().symbol(member);